File: cost/iona/orbacus/java/main/TestIntfContext_impl.java
// **********************************************************************
//
// Copyright (c) 2001
// IONA Technologies, Inc.
// Waltham, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

package ORBTest;

import org.omg.CORBA.*;
import org.omg.PortableServer.*;
import ORBTest_Context.*;

final class TestIntfContext_impl extends ORBTest_Context.IntfPOA
{
    private POA m_poa;

    public
    TestIntfContext_impl(
    	POA poa
    )
    {
	m_poa = poa;
    }

    public synchronized String[]
    opContext(
	String pattern,
	Context ctx
    )
    {
	NVList values = ctx.get_values("", 0, pattern);
	
	int len = values.count();
	String[] result = new String[len * 2];
	for(int i = 0 ; i < len ; i++)
	{
	    try
	    {
		String s = values.item(i).value().extract_string();
		result[i * 2] = values.item(i).name();
		result[i * 2 + 1] = s;
	    }
	    catch(org.omg.CORBA.Bounds ex)
	    {
	    }
	}
	
	return result;
    }

    public org.omg.PortableServer.POA
    _default_POA()
    {
	return m_poa;
    }
}

File: cost/iona/orbacus/cpp/main/TestIntfContext_impl.cpp
// **********************************************************************
//
// Copyright (c) 2001
// IONA Technologies, Inc.
// Waltham, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

#include <includes.h>

#include <TestIntfContext_impl.h>

#include <limits.h>

using namespace CORBA;

TestIntfContext_impl::TestIntfContext_impl()
{
}

ORBTest_Context::StringSequence*
TestIntfContext_impl::opContext(
    const char* pattern,
    Context_ptr ctx
)
{
#if defined(__GNUC__) && defined(__OPTIMIZE__) // COMPILERFIX
    volatile int dummy = 1;
    dummy++;
#endif

    NVList_var values;
    ctx->get_values("", 0, pattern, values.out());

    ULong len = values->count();
    ORBTest_Context::StringSequence* result = (
        new ORBTest_Context::StringSequence(len * 2)
    );
    result->length(len * 2);
    for (ULong i = 0; i < len; i++)
    {
	const char* s;
	*(values->item(i)->value()) >>= s;
	(*result)[i * 2] = values->item(i)->name();
	(*result)[i * 2 + 1] = string_dup(s);
    }

    return result;
}
